File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        
        int[] reference = new int[nums.length + 1];
        reference[1] = nums[0];
        
        for (int i = 2; i <= nums.length; i++) {
            reference[i] = Math.max(reference[i - 1], reference[i - 2] + nums[i - 1]);
        }
        
        return reference[nums.length];
    }
}

File: 0198-house-robber/0198-house-robber.cs
public class Solution {
    public int Rob(int[] nums) {
        List<int> dp = Enumerable.Repeat(0, nums.Length + 1).ToList();
        dp[1] = nums[0];
        for (int i = 2; i <= nums.Length; i++) {
            dp[i] = Math.Max(dp[i - 2] + nums[i - 1], dp[i - 1]);
        }
        return dp[nums.Length];
    }
}

